bin/genesis

```diff
@@ -90,6 +133,34 @@ setup() {
 		exit 3
 	fi
 
+	if [[ -f ${DEPLOYMENT_ROOT}/.genesis_deps ]]; then
+		errors=""
+		while IFS='' read -r line || [[ -n "$line" ]]; do
+			cmd=$(echo $line | cut -d ":" -f1)
+			version=$(echo $line | cut -d ":" -f2- | auto_sed 's/^ +//' | auto_sed 's/ +$//')
```

Comment:
My original thought here was to use native string handling routines in bash, to avoid the dep on sed (and the auto_sed function).  Here's what I ended up with needing:

```
cmd=${line%%:*}; cmd="${cmd%"${cmd##*[![:space:]]}"}"; cmd="${cmd#"${cmd%%[![:space:]]*}"}"
ver=${line##*:}; ver="${ver%"${ver##*[![:space:]]}"}"; ver="${ver#"${ver%%[![:space:]]*}"}"
```

(note that I renamed `version` to `ver` to line up the columns)

Not sure how I feel about this.  Maybe a `trim` function?